src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1022,7 +1022,11 @@ void QgsOgrProviderUtils::setRelevantFields( OGRLayerH ogrLayer, int fieldCount,
       if ( !fetchAttributes.contains( i ) )
       {
         // add to ignored fields
-        ignoredFields.append( OGR_Fld_GetNameRef( OGR_FD_GetFieldDefn( featDefn, firstAttrIsFid ? i - 1 : i ) ) );
+        const char *fieldName = OGR_Fld_GetNameRef( OGR_FD_GetFieldDefn( featDefn, firstAttrIsFid ? i - 1 : i ) );
+        if ( qstrcmp( fieldName, "orig_ogc_fid" ) != 0 )
```

Comment:
Shouldn't we use a constant name instead of hardcoding "orig_ogc_fid" in many places ?

Reply:
Done - don't know however if this should be a publicly accessible constant of OgrProvider or just an internal symbol. Currently `test_provider_ogr_sqlite.py` still hardcodes `orig_ogc_fid`.

Reply:
I wouldn't feel very confortable to expose it publicly as it is really an implementation detail. It is OK if we have a test that hardcodes it.